packages/libs/eda/src/lib/core/components/computations/plugins/differentialabundance.tsx

```diff
@@ -0,0 +1,276 @@
+import {
+  AnalysisState,
+  useCollectionVariables,
+  useStudyMetadata,
+} from '../../..';
+import { VariableDescriptor } from '../../../types/variable';
+import { volcanoPlotVisualization } from '../../visualizations/implementations/VolcanoPlotVisualization';
+import { ComputationConfigProps, ComputationPlugin } from '../Types';
+import { isEqual, partial } from 'lodash';
+import { useConfigChangeHandler, assertComputationWithConfig } from '../Utils';
+import * as t from 'io-ts';
+import { Computation } from '../../../types/visualization';
+import SingleSelect from '@veupathdb/coreui/lib/components/inputs/SingleSelect';
+import { useFindEntityAndVariable } from '../../../hooks/workspace';
+import { useMemo } from 'react';
+import { ComputationStepContainer } from '../ComputationStepContainer';
+import { sharedConfigCssStyles } from './abundance';
+import VariableTreeDropdown from '../../variableTrees/VariableTreeDropdown';
+import { ValuePicker } from '../../visualizations/implementations/ValuePicker';
+import { useToggleStarredVariable } from '../../../hooks/starredVariables';
+
+/**
+ * Differential abundance
+ *
+ * The differential abundance app is used to find taxa, genes, pathways, etc. that
+ * are more abundant in one group of samples than another. This app takes in a continuous variable
+ * collection (for example, abundance of 100 Species for all samples) as well as a way to split the samples
+ * into two groups (for example, red hair and green hair). The computation then returns information on how
+ * differentially abundance each item (taxon, gene, etc.) is between the two groups. See VolcanoPlotDataPoint
+ * for more details. Importantly, the returned data lives outside the variable tree because each returned
+ * data point corresponds to an item (taxon, gene, etc.).
+ *
+ * Currently the differential abundance app will be implemented with only a volcano visualization. Plans for
+ * the future of this app include a lefse diagram, tables of results, adding new computation methods, and
+ * strategies to create user-defined collections from the output of the computation.
+ */
+
+export type DifferentialAbundanceConfig = t.TypeOf<
+  typeof DifferentialAbundanceConfig
+>;
+
+const Comparator = t.type({
+  variable: VariableDescriptor,
+  groupA: t.array(t.string),
+  groupB: t.array(t.string),
+});
```

Comment:
I noticed the "Generate" button is still enabled when Group A and Group B values aren't selected. I think what's happening is that they are being set to empty arrays when a comparator var is selected, so it passes the `isConfigurationValid` check that [is done in the viz container component](https://github.com/VEuPathDB/web-monorepo/blob/496354de2ea8e6c850978e1f528ca6409ef71257/packages/libs/eda/src/lib/core/components/visualizations/VisualizationsContainer.tsx#L548).

If we set `groupA` and `groupB` to null instead of an empty array, we can get the disabling as desired. But this results in a TS error and some other strange behavior: the first selection made in the picker gets removed when first rendered. Doesn't seem to happen thereafter. 😵

Reply:
aha yes! Okay i will work on this now! Thank you!

And if you find one more console log or unused import in this PR your Friday beer is on me!

Reply:
(dont look yet though i just found one)

Reply:
alright @jernestmyers all fixed! I ended up defining a custom `isConfigurationValid` function. That was a good clue!

Reply:
Ah yes, good idea! Looks great.